modules/raft/src/main/java/org/apache/ignite/raft/jraft/storage/impl/LogManagerImpl.java

```diff
@@ -821,9 +824,10 @@ public long getTerm(final long index) {
             if (index > this.lastLogIndex || index < this.firstLogIndex) {
                 return 0;
             }
-            final LogEntry entry = getEntryFromMemory(index);
```

Comment:
So, there is already some kind of a cache? Why didn't it work?

Reply:
This cache is being cleared all the time, it doesn't have enough data in it

Reply:
Why is it being cleared all the time? Maybe we should have solved this problem instead?

Reply:
I believe this should be a separate discussion, I'm not ready to make such changes right now. Should I create a new Jira for investigation of cache misses in `logsInMemory`?

Reply:
I'm not sure either....

Reply:
Alright, let's do that in the future, I'll keep this issue in mind